components/org.wso2.carbon.identity.organization.resource.sharing.policy.management/src/main/java/org/wso2/carbon/identity/organization/resource/sharing/policy/management/ResourceSharingPolicyHandlerServiceImpl.java

```diff
@@ -0,0 +1,142 @@
+
+    @Override
+    public int addResourceSharingPolicy(ResourceSharingPolicy resourceSharingPolicy)
+            throws ResourceSharingPolicyMgtServerException {
+
+        return resourceSharingPolicyHandlerDAO.addResourceSharingPolicyRecord(resourceSharingPolicy);
+    }
+
+    @Override
+    public List<ResourceSharingPolicy> getResourceSharingPolicies(List<String> organizationIds)
+            throws ResourceSharingPolicyMgtServerException {
+
+        return resourceSharingPolicyHandlerDAO.getResourceSharingPolicies(organizationIds);
+    }
+
+    @Override
+    public Map<ResourceType, List<ResourceSharingPolicy>> getResourceSharingPoliciesGroupedByResourceType(
+            List<String> organizationIds) throws ResourceSharingPolicyMgtServerException {
+
+        return resourceSharingPolicyHandlerDAO.getResourceSharingPoliciesGroupedByResourceType(organizationIds);
+    }
+
+    @Override
+    public Map<String, List<ResourceSharingPolicy>> getResourceSharingPoliciesGroupedByPolicyHoldingOrgId(
+            List<String> organizationIds) throws ResourceSharingPolicyMgtServerException {
+
+        return resourceSharingPolicyHandlerDAO.getResourceSharingPoliciesGroupedByPolicyHoldingOrgId(organizationIds);
+    }
+
+    @Override
+    public boolean deleteResourceSharingPolicyRecordById(Integer resourceSharingPolicyId, String permittedOrgId)
+            throws ResourceSharingPolicyMgtServerException {
+
+        return resourceSharingPolicyHandlerDAO.deleteResourceSharingPolicyRecordById(resourceSharingPolicyId,
+                permittedOrgId);
+    }
+
+    @Override
+    public boolean deleteResourceSharingPolicyByResourceTypeAndId(ResourceType resourceType, String resourceId,
+                                                                  String permittedOrgId)
+            throws ResourceSharingPolicyMgtServerException {
+
+        return resourceSharingPolicyHandlerDAO.
+                deleteResourceSharingPolicyByResourceTypeAndId(resourceType, resourceId, permittedOrgId);
+    }
+
+    @Override
+    public boolean addSharedResourceAttributes(List<SharedResourceAttribute> sharedResourceAttributes)
+            throws ResourceSharingPolicyMgtServerException {
+
+        return resourceSharingPolicyHandlerDAO.addSharedResourceAttributes(sharedResourceAttributes);
+    }
+
+    @Override
+    public List<SharedResourceAttribute> getSharedResourceAttributesBySharingPolicyId(Integer resourceSharingPolicyId)
+            throws ResourceSharingPolicyMgtServerException, DataAccessException {
+
+        return resourceSharingPolicyHandlerDAO.getSharedResourceAttributesBySharingPolicyId(resourceSharingPolicyId);
+    }
+
+    @Override
+    public List<SharedResourceAttribute> getSharedResourceAttributesByType(SharedAttributeType attributeType)
+            throws ResourceSharingPolicyMgtServerException {
+
+        return resourceSharingPolicyHandlerDAO.
+                getSharedResourceAttributesByType(attributeType);
```

Comment:
```suggestion
        return resourceSharingPolicyHandlerDAO.getSharedResourceAttributesByType(attributeType);
```
Hope the character limit issue is not there.
Fix other similar places as well

Reply:
Addressed and resolved in commit: [6153882](https://github.com/wso2-extensions/identity-organization-management/pull/416/commits/61538829b53d28498f1f0529a53b56c742574c84)